### ISSUE TITLE
Runtime Error on Jetson TX2

### ISSUE THREAD

**Issue by laxnpander** [CESNET/GPUJPEG#54]
Hey,

I installed GPUJPEG on a Jetson TX2 board. However, when I try to use the library in my own code the following error is thrown:

```
[GPUJPEG] [Error] Failed to initialize CUDA device.
terminate called after throwing an instance of 'std::runtime_error'
  what():  
Aborted (core dumped)
```

The cuda samples are running perfectly fine, so CUDA should be set up correctly. Any suggestions?

**Comment by MartinPulec** [comment]
Hi Alexander, hard to say without any details. I've pushed a small change that at least prints CUDA error in this place.

Also, would it be possible to try out the project sample application, eg.:

    gpujpeg -e -c ycbcr-bt709 -f 422-u8-p1020 -s 1920x1080 colors/camera_bt709_422.yuv out.jpg

If the above works, would it be possible to provide a (ideally minimal) working example exhibiting the problem?

**Comment by laxnpander** [comment]
@MartinPulec: Hey Martin, jeah sorry I know. Information is a bit thin for debugging. I already got some prints for the error myself, and it throws error 30... which is unknown error on the Nvidia side. Also very helpful. I investigated further and it might have to do with some corrupted flash / driver issue. So nothing on your side. Though, it is weird that I am able to run all the CUDA examples just fine. But I'll let you know, when I found out more.

**Comment by MartinPulec** [comment]
Ok then, thanks for the info. Let me know if you find something then.

Some people are using GPUJPEG on Jetson (and IIRC also Jetson TX2), but I don't have any (or similar) to test, so I'd definitely need more info about the potential problem.

**Comment by laxnpander** [comment]
@MartinPulec: Hey, I found a way to use hardware accelerated jpeg compression not bound to CUDA on the Jetson TX2. So I put your library aside for now. But in case I return to it and can investigate more, what is going wrong, I will report back here. For now I will close this issue.

**Comment by DenisN03** [comment]
Hello @laxnpander! What image encoding solution have you come to use?

**Comment by laxnpander** [comment]
@DenisN03: Uuuuh, I have to remember what I did a year ago. I think the Tegra comes with an FPGA or processor solely for compression which I utilised. Here is my code snippet to do that:

https://github.com/laxnpander/ROS_Tegra_Utilities/blob/master/src/ros_tegra_utilities_lib/tegra_utilities.cpp

But I am not exactly sure it was working flawlessly. I remember seeing some visual artifacts as in a faulty transmission (image half black or distorted) between my camera and the final result. I never investigated where it came from. In case you are compressing images from disk it might be less of a problem.
